src/main/java/org/jboss/msc/service/ServiceContainerImpl.java

```diff
@@ -744,9 +744,9 @@ void apply(ServiceBuilderImpl<?> builder) {
         final ServiceControllerImpl<T> instance = new ServiceControllerImpl<T>(serviceBuilder.getServiceValue(),
                 dependencies, valueInjectionArray, outInjectionArray, primaryRegistration, aliasRegistrations,
                 serviceBuilder.getMonitors(), serviceBuilder.getListeners(), serviceBuilder.getLifecycleListeners(), serviceBuilder.getParent());
+        serviceValue.setValue(instance);
```

Comment:
Since we're making changes here, there should be a comment to the effect of "this will never throw an exception".